### PR TITLE
Use composer to install PHPCS and friends on Travis

### PR DESCRIPTION
**Pull request by grappler**
#### Changes proposed in this Pull Request:
Use composer instead of git to install WPCS and PHPCompatibility. This was recommend in a previous PR https://github.com/Automattic/_s/pull/1183#issuecomment-315844661

Changing the flow to https://github.com/Automattic/_s/pull/1267/commits/a0d2892cadbc290ac7e4eb507c7e5cfc55202a53

Only use the development version of PHPCompatibility. With this change we would use the stable version of PHPCS.  Due to an update to the PHPCS master the checks are failing. I would like to the necessary changes to happen along side the PHPCS and WPCS updates instead of having to do them with other changes.

#### Related issue(s):
#1267

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -48,19 +48,13 @@ matrix:
...
-    - export WPCS_DIR=/tmp/wpcs
-    - export PHPCOMPAT_DIR=/tmp/phpcompatibility
-    # Install CodeSniffer for WordPress Coding Standards checks.
-    - if [[ "$SNIFF" == "1" ]]; then git clone -b master --depth 1 https://github.com/squizlabs/PHP_CodeSniffer.git $PHPCS_DIR; fi
-    # Install WordPress Coding Standards.
-    - if [[ "$SNIFF" == "1" ]]; then git clone -b 0.14.1 --depth 1 https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards.git $WPCS_DIR; fi
-    # Install PHP Compatibility sniffs.
-    - if [[ "$SNIFF" == "1" ]]; then git clone -b master --depth 1 https://github.com/wimg/PHPCompatibility.git $PHPCOMPAT_DIR; fi
+    # Install WordPress Coding Standards & PHP Compatibility sniffs.
+    - if [[ "$SNIFF" == "1" ]]; then composer global require wp-coding-standards/wpcs:~0.14.1 wimg/php-compatibility:dev-master --prefer-stable --no-interaction --no-suggest; fi
```
Comment (jrfnl) [inline .travis.yml]: Please add `dealerdirect/phpcodesniffer-composer-installer:^0.4.3`. This is a Composer plugin which will sort out the installed paths automatically and is suggested for that reason by both the standards.

File: .travis.yml [hunk 2]
```diff
@@ -48,19 +48,13 @@ matrix:
...
+    # Set composer PATH.
+    - if [[ "$SNIFF" == "1" ]]; then export PATH=$HOME/.composer/vendor/bin:$PATH; fi
     # Set install path for PHPCS sniffs.
     # @link https://github.com/squizlabs/PHP_CodeSniffer/blob/4237c2fc98cc838730b76ee9cee316f99286a2a7/CodeSniffer.php#L1941
-    - if [[ "$SNIFF" == "1" ]]; then $PHPCS_DIR/bin/phpcs --config-set installed_paths $WPCS_DIR,$PHPCOMPAT_DIR; fi
+    - if [[ "$SNIFF" == "1" ]]; then phpcs --config-set installed_paths $HOME/.composer/vendor/wp-coding-standards/wpcs,$HOME/.composer/vendor/wimg/php-compatibility; fi
```
Comment (jrfnl) [inline .travis.yml]: If the above mentioned Composer plugin is added, this line can be removed completely.

**Comment by jrfnl** [comment]
Interesting approach doing this without a `composer.json`, though I do think it would be better to have a `composer.json` file, if only to document which versions of the various libraries are being used in a more obvious way than hidden away in a Travis script.

Regarding the build failure: this has to do with a change in PHPCS - see https://github.com/squizlabs/PHP_CodeSniffer/issues/1539. Please note that to version which would currently be used `3.2.3` is buggy regarding the new PHPCS annotations.

**Comment by Ismail-elkorchi** [comment]
Closing as composer was added in https://github.com/Automattic/_s/pull/1386.

**Pull request closed without merging**